plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/StandardFacetInstallDelegate.java

```
@@ -46,22 +47,49 @@ public void execute(IProject project,
     installAppEngineRuntimes(project);
   }
 
-  private void installAppEngineRuntimes(final IProject project) {
+  private void installAppEngineRuntimes(final IProject project) throws CoreException {
+    final IFacetedProject facetedProject = ProjectFacetsManager.create(project);
+
     // Modifying targeted runtimes while installing/uninstalling facets is not allowed,
     // so schedule a job as a workaround.
     Job installJob = new Job("Install App Engine runtimes in " + project.getName()) {
       @Override
       protected IStatus run(IProgressMonitor monitor) {
+        // https://github.com/GoogleCloudPlatform/google-cloud-eclipse/issues/1155
+        // Wait until the first ConvertJob installs the JSDT facet.
+        try {
+          IProjectFacet jsdtFacet = ProjectFacetsManager.getProjectFacet("wst.jsdt.web");
+          for (int times = 0;
+              times < 100 && !facetedProject.isFixedProjectFacet(jsdtFacet);
+              times++) {
+            try {
+              // To prevent going into the SLEEPING state, don't use "Job.schedule(100)".
+              Thread.sleep(100 /* ms */);
+            } catch (InterruptedException ex) {}
```

empty catch blocks in non-test code require a comment